llvm/CMakeLists.txt

```diff
@@ -707,8 +707,15 @@ endif( LLVM_USE_PERF )
 set(LLVM_USE_SANITIZER "" CACHE STRING
   "Define the sanitizer used to build binaries and tests.")
 option(LLVM_OPTIMIZE_SANITIZED_BUILDS "Pass -O1 on debug sanitizer builds" ON)
+
+if( CMAKE_C_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_ID MATCHES "Clang" )
+  set(LLVM_UBSAN_FLAGS_CLANG "-fno-sanitize=function")
+else()
+  # gcc doesn't know about -fsanitize=function
+  set(LLVM_UBSAN_FLAGS_CLANG)
```

Comment:
Current GCC does support `-fsanitize=`: https://clang.godbolt.org/z/nT3bPW766

Something else must be wrong with your build. Maybe it's an old version?